components-core/src/main/java/com/adyen/checkout/components/core/internal/analytics/data/remote/AnalyticsTrackRequestProvider.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2024 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by ararat on 4/3/2024.
+ */
+
+package com.adyen.checkout.components.core.internal.analytics.data.remote
+
+import com.adyen.checkout.components.core.internal.analytics.AnalyticsEvent
+import com.adyen.checkout.components.core.internal.analytics.AnalyticsPlatformParams
+import com.adyen.checkout.components.core.internal.data.model.AnalyticsTrackInfo
+import com.adyen.checkout.components.core.internal.data.model.AnalyticsTrackLog
+import com.adyen.checkout.components.core.internal.data.model.AnalyticsTrackRequest
+
+internal class AnalyticsTrackRequestProvider {
+
+    operator fun invoke(
```

Comment:
is there a specific reason this being an operator function?

Reply:
Not necessarily, but this class is supposed to have 1 responsibility and this sort of reflects that